From 33576855887b196c966ebe0fec4972e233e12625 Mon Sep 17 00:00:00 2001
From: Murali Karicheri <m-karicheri2@ti.com>
Date: Wed, 3 Feb 2016 14:10:19 -0500
Subject: [PATCH 3/6] net: netcp: pa: add missing of_node_put() on bail out

for_each_child_of_node() calls of_node_put() only for previous node.
So if we break from the loop, caller needs to call of_node_put() on
the current child.

Signed-off-by: Murali Karicheri <m-karicheri2@ti.com>
---
 drivers/net/ethernet/ti/netcp_pa_core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/ti/netcp_pa_core.c b/drivers/net/ethernet/ti/netcp_pa_core.c
index 0df7656..00bafc0 100644
--- a/drivers/net/ethernet/ti/netcp_pa_core.c
+++ b/drivers/net/ethernet/ti/netcp_pa_core.c
@@ -254,6 +254,7 @@ static int pa_core_get_dt_bindings(struct pa_core_device *core_dev,
 	return ret;
 
 release_of_node:
+	of_node_put(child);
 	of_node_put(clusters);
 	return ret;
 }
-- 
1.9.1

